Manual testing falsely trips ERR_MODULE_NOT_FOUND rethrow logic?

It seems that manual testing with something [like this](https://github.com/NullVoxPopuli/ember-apply/blob/main/packages/ember/tailwind/manual-test.sh):
```
node /some/path/ember-apply/src/cli/index.js tailwind
```

can cause this logic to fail:
https://github.com/NullVoxPopuli/ember-apply/blob/2309aea575fd4bf8ed5cf8970303e9486df67653/ember-apply/src/cli/resolve.js#L274-L281

By "fail" I mean that the error should've been swallowed as it merely indicates that the module being imported was not a path.
e.g. In the case below (attempting to run local ember-apply script in ember-cli-plotly project dir) it failed because the stack trace included "tryResolve ... ember-apply/src/cli/resolve.js"

```
Cannot find module 'C:\Users\jacob\Documents\ember-cli-plotly\tailwind' imported from C:\Users\jacob\Documents\ember-apply\ember-apply\src\cli\resolve.js
Error [ERR_MODULE_NOT_FOUND]: Cannot find module 'C:\Users\jacob\Documents\ember-cli-plotly\tailwind' imported from C:\Users\jacob\Documents\ember-apply\ember-apply\src\cli\resolve.js
    at new NodeError (node:internal/errors:372:5)
    at finalizeResolution (node:internal/modules/esm/resolve:437:11)
    at moduleResolve (node:internal/modules/esm/resolve:1009:10)
    at defaultResolve (node:internal/modules/esm/resolve:1218:11)
    at ESMLoader.resolve (node:internal/modules/esm/loader:580:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:294:18)
    at ESMLoader.import (node:internal/modules/esm/loader:380:22)
    at importModuleDynamically (node:internal/modules/esm/translators:106:35)
    at importModuleDynamicallyCallback (node:internal/process/esm_loader:35:14)
    at tryResolve (file:///C:/Users/jacob/Documents/ember-apply/ember-apply/src/cli/resolve.js:272:27) {
  code: 'ERR_MODULE_NOT_FOUND'
}
```

It's also admittedly possible that I broke something (I'm working off the fork in https://github.com/NullVoxPopuli/ember-apply/pull/262, which [touched the resolve logic](https://github.com/NullVoxPopuli/ember-apply/pull/262/files#diff-eef8ad60356cd9c0197f670fa95d7448bc929862fbf9acc5ea8e28d7893df0d6)) or am doing something wrong.

Hmm, looks like there're still some Windows issues that I should look at later. It seems this particular issue is not restricted to manual test runs but also happens when I run via `npx` (tested with ember-apply v2.2.3)

I tried manually editing `ember-apply\src\cli\resolve.js` (in npx cache) to swallow the error like this:
 
```js
if (error.code === 'ERR_MODULE_NOT_FOUND') {
  /**
   * If *we* make a mistake, don't swallow the error
   */
  if (!error.message.includes('ember-apply/src/cli/')) {
    console.log(chalk.yellow('\nActually, do swallow this one'), error);
    //throw error;
  }
}
```

However, I still hit another problem that appears due to Windows absolute path not getting prefixed with `file://` (maybe caused [here](https://github.com/NullVoxPopuli/ember-apply/blob/426896e91e26216e9f25d83cb5dd72b203438446/ember-apply/src/cli/resolve.js#L169)). 

<details><summary>Output</summary>

```
user@computer MINGW64 ~/tmp/empty-app (main)
$ npx ember-apply@latest --verbose tailwind
Detected CWD: C:\Users\user\tmp\empty-app
⠋ Locating feature: tailwind
Checking file://C:\Users\user\tmp\empty-app\tailwind

Actually, do swallow this one Error [ERR_MODULE_NOT_FOUND]: Cannot find module 'C:\Users\user\tmp\empty-app\tailwind' imported from C:\Users\user\AppData\Local\npm-cache\_npx\5066d7421dc769ed\node_modules\ember-apply\src\cli\resolve.js
    at new NodeError (node:internal/errors:372:5)
    at finalizeResolution (node:internal/modules/esm/resolve:437:11)
    at moduleResolve (node:internal/modules/esm/resolve:1009:10)
    at defaultResolve (node:internal/modules/esm/resolve:1218:11)
    at ESMLoader.resolve (node:internal/modules/esm/loader:580:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:294:18)
    at ESMLoader.import (node:internal/modules/esm/loader:380:22)
    at importModuleDynamically (node:internal/modules/esm/translators:106:35)
    at importModuleDynamicallyCallback (node:internal/process/esm_loader:35:14)
    at tryResolve (file:///C:/Users/user/AppData/Local/npm-cache/_npx/5066d7421dc769ed/node_modules/ember-apply/src/cli/resolve.js:269:27) {
  code: 'ERR_MODULE_NOT_FOUND'
}
Error [ERR_MODULE_NOT_FOUND]: Cannot find module 'C:\Users\user\tmp\empty-app\tailwind' imported from C:\Users\user\AppData\Local\npm-cache\_npx\5066d7421dc769ed\node_modules\ember-apply\src\cli\resolve.js
Checking file://C:\Users\user\tmp\empty-app\tailwind\index.js

Actually, do swallow this one Error [ERR_MODULE_NOT_FOUND]: Cannot find module 'C:\Users\user\tmp\empty-app\tailwind\index.js' imported from C:\Users\user\AppData\Local\npm-cache\_npx\5066d7421dc769ed\node_modules\ember-apply\src\cli\resolve.js
    at new NodeError (node:internal/errors:372:5)
    at finalizeResolution (node:internal/modules/esm/resolve:437:11)
    at moduleResolve (node:internal/modules/esm/resolve:1009:10)
    at defaultResolve (node:internal/modules/esm/resolve:1218:11)
    at ESMLoader.resolve (node:internal/modules/esm/loader:580:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:294:18)
    at ESMLoader.import (node:internal/modules/esm/loader:380:22)
    at importModuleDynamically (node:internal/modules/esm/translators:106:35)
    at importModuleDynamicallyCallback (node:internal/process/esm_loader:35:14)
    at tryResolve (file:///C:/Users/user/AppData/Local/npm-cache/_npx/5066d7421dc769ed/node_modules/ember-apply/src/cli/resolve.js:269:27) {
  code: 'ERR_MODULE_NOT_FOUND'
}
Error [ERR_MODULE_NOT_FOUND]: Cannot find module 'C:\Users\user\tmp\empty-app\tailwind\index.js' imported from C:\Users\user\AppData\Local\npm-cache\_npx\5066d7421dc769ed\node_modules\ember-apply\src\cli\resolve.js
Checking https://cdn.skypack.dev/@ember-apply/tailwind
Error [ERR_UNSUPPORTED_ESM_URL_SCHEME]: Only URLs with a scheme in: file, data are supported by the default ESM loader. Received protocol 'https:'
Checking https://cdn.skypack.dev/tailwind
Error [ERR_UNSUPPORTED_ESM_URL_SCHEME]: Only URLs with a scheme in: file, data are supported by the default ESM loader. Received protocol 'https:'
ℹ Skypack unavailable, downloading from npm
Checking C:\Users\user\AppData\Local\Temp\ember-apply-runtime---9hy9ww\index.js
Error [ERR_UNSUPPORTED_ESM_URL_SCHEME]: Only URLs with a scheme in: file, data are supported by the default ESM loader. On Windows, absolute paths must be valid file:// URLs. Received protocol 'c:'
✖ applyable feature must be an object. got: undefined
AssertionError [ERR_ASSERTION]: applyable feature must be an object. got: undefined
    at getApplyable (file:///C:/Users/user/AppData/Local/npm-cache/_npx/5066d7421dc769ed/node_modules/ember-apply/src/cli/index.js:89:3)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async run (file:///C:/Users/user/AppData/Local/npm-cache/_npx/5066d7421dc769ed/node_modules/ember-apply/src/cli/index.js:66:21)
    at async Object.handler (file:///C:/Users/user/AppData/Local/npm-cache/_npx/5066d7421dc769ed/node_modules/ember-apply/src/cli/index.js:44:9) {
  generatedMessage: false,
  code: 'ERR_ASSERTION',
  actual: false,
  expected: true,
  operator: '=='
}
```

</details>

--- a/ember-apply/src/cli/resolve.js
+++ b/ember-apply/src/cli/resolve.js
@@ -166,7 +166,17 @@ async function downloadFromNpm(options) {
     spinner.stop();
   }
 
-  return await tryResolve(path.join(dir, main), options);
+  let url = path.join(dir, main);
+
+  if (os.platform() === 'win32') {
+    url = convertWindowsAbsolutePathToFileUrl(url);
+
+    if (options.verbose) {
+      console.info(chalk.gray(`Rewriting as file URL for Windows compatibility --> ${url}`));
+    }
+  }
+
+  return await tryResolve(url, options);
 }
 
 /**
@@ -205,6 +215,20 @@ async function resolvePackage(options) {
 }
 
 /**
+ * On Windows, the ESM loader requires that absolute paths must be valid file:// URLs.
+ * Absolute paths (e.g. with drive letters like "C:") can lead to ERR_UNSUPPORTED_ESM_URL_SCHEME
+ * When this function detects strings that begin with "drive letters" it returns a new
+ * path string that is prefixed with `file://`.
+ * Otherwise, it returns its argument without modification.
+ *
+ * @param {string} path
+ * @returns {string} adjustedPath
+ */
+function convertWindowsAbsolutePathToFileUrl(p) {
+  return p.match(/^[a-zA-Z]:/) ? `file://${p}` : p;
+}
+
+/**
  * @typedef {object} LocalModuleInfo
  * @property {string | undefined} path
  * @property {UnknownModule | undefined} module
@@ -226,11 +250,11 @@ async function resolvePath(options) {
     path.resolve(name),
     // local, but without index.js
     path.resolve(name, 'index.js'),
-  ].map((p) => {
-    // On Windows, the ESM loader requires that absolute paths must be valid file:// URLs.
-    // Absolute paths (e.g. with drive letters like "C:") can lead to ERR_UNSUPPORTED_ESM_URL_SCHEME
-    return p.match(/^[a-zA-Z]:/) ? `file://${p}` : p;
-  });
+  ];
+
+  if (os.platform() === 'win32') {
+    potentialLocations = potentialLocations.map(convertWindowsAbsolutePathToFileUrl);
+  }
 
   let resolvedModule;
   let resolvedPath;
@@ -273,7 +297,7 @@ async function tryResolve(url, options = {}) {
       /**
        * If *we* make a mistake, don't swallow the error
        */
-      if (!error.message.includes('ember-apply/src/cli/')) {
+      if (!error.message.includes(path.join('ember-apply', 'src', 'cli'))) {
         throw error;
       }
     }

